CONTRIBUTING.md

```diff
@@ -1,31 +1,129 @@
+
+### Setting up your local development environment
+
+1. Install a valid [Dart SDK](https://dart.dev/get-dart) in your local environment. Compatible Dart SDK versions with test optimizer can be found [here](https://github.com/VeryGoodOpenSource/very_good_cli/blob/66c2a5e847ba4db6a9cb212e2b08e2af9f9e70d2/pubspec.yaml#L7). If you have Flutter installed, you likely have a valid Dart SDK version already installed.
+
+2. Install all Very Good CLI's dependencies:
+
+```sh
+# 📂 Get project dependencies recursively with Very Good CLI
+very_good packages get -r
+
+# Or get project dependencies manually
+dart pub get && cd bricks/test_optimizer && dart pub get && cd ../../
+```
+
+3. Run all Very Good CLI tests:
+
+```sh
+# 💻 Run Very Good CLI's unit tests
+dart test
+```
+
+If some tests do not pass out of the box, please submit an [issue](https://github.com/VeryGoodOpenSource/very_good_cli/issues/new/choose).
+
+4. Install your own version of Very Good CLI in your local environment:
+
+```sh
+# 🚀 Activate your own local version of Very Good CLI
+dart pub global activate --source path .
+```
+
+### Creating a Pull Request
+
+Before creating a Pull Request please:
+
+1. [Fork](https://docs.github.com/en/get-started/quickstart/contributing-to-projects) the [GitHub repository](https://github.com/VeryGoodOpenSource/very_good_cli) and create your branch from `main`:
+
+```sh
+# 🪵 Branch from `main`
+git branch <branch-name>
+git checkout <branch-name>
+```
+
+Where `<branch-name>` is an appropriate name describing your change.
+
+2. Install dependencies:
+
+```sh
+# 📂 Get project dependencies recursively with Very Good CLI
+very_good packages get -r
+
+# Or get project dependencies manually
+dart pub get && cd bricks/test_optimizer && dart pub get && cd ../../
+```
+
+3. Ensure you have a meaningful [semantic][conventional_commits_link] commit message.
+
+4. Add tests! Pull Requests without 100% test coverage will **not** be approved. If you're unsure on how to do so watch our [Testing Fundamentals Course](https://www.youtube.com/watch?v=M_eZg-X789w&list=PLprI2satkVdFwpxo_bjFkCxXz5RluG8FY).
+
+5. Ensure the existing test suite passes locally:
+
+```sh
+# 🧪 Run Very Good CLI's unit test
+dart test
+```
+
+6. Format your code:
+
+```sh
+# 🧼 Run Dart's formatter
+dart format .
+```
+
+7. Analyze your code:
+
+```sh
+# 🔍 Run Dart's analyzer
+dart analyze --fatal-infos --fatal-warnings .
+```
+
+Some analysis issues may be fixed automatically with:
+
+```sh
+# Automatically fix analysis issues that have associated automated fixes
+dart fix --apply
+```
+
+> **Note**: Our repositories use [Very Good Analysis](https://github.com/VeryGoodOpenSource/very_good_analysis). 
+
+8. Create the Pull Request with a meaningful description, linking to the original issue where possible.
+
+9. Verify that all [status checks](https://github.com/VeryGoodOpenSource/very_good_cli/actions/) are passing.
```

Comment:
If it is a fork, the actions wont live on our repo right? Also the actions will only run if we approve them to run for first-time contributors

Reply:
I've changed this, let me know if you're happy with the new change by resolving this comment.